### PR TITLE
Allowing sharing on whatsapp

### PR DESCRIPTION
**Pull request by prdm0**
I find it very useful to share the link to the url that hosts the book via whatsapp. I am sure that this feature will appeal to many people.

### CODE REVIEW

**Comment by prdm0** [comment]
An example of a website working with the share icon via whatsapp: <https://prdm0.github.io/regimento_interno/>.

File: inst/resources/gitbook/css/style.css [hunk 1]
```diff
@@ -6,7 +6,7 @@
...
... .book-body .navigation.navigation-prev{left:0}@media (max-width:1240px){.book .book-body .navigation{position:static;top:auto;max-width:50%;width:50%;display:inline-block;float:left}.book .book-body .navigation.navigation-unique{max-width:100%;width:100%}}.book .book-body .page-wrapper .page-inner section.glossary{margin-bottom:40px}.book .book-body .page-wrapper .page-inner section.glossary h2 a,.book .book-body .page-wrapper .page-inner section.glossary h2 a:hover{color:inherit;text-decoration:none}.book .book-body .page-wrapper .page-inner section.glossary .glossary-index{list-style:none;margin:0;padding:0}.book .book-body .page-wrapper .page-inner section.glossary .glossary-index li{display:inline;margin:0 8px;white-space:nowrap}*{-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box;-webkit-overflow-scrolling:touch;-webkit-tap-highlight-color:transparent;-webkit-text-size-adjust:none;-webkit-touch-callout:none}a{text-decoration:none}body,html{height:100%}html{font-size:62.5%}body{text-rendering:optimizeLegibility;font-smoothing:antialiased;font-family:"Helvetica Neue",Helvetica,Arial,sans-serif;font-size:14px;letter-spacing:.2px;text-size-adjust:100%}
```
Comment (yihui) [inline inst/resources/gitbook/css/style.css:9]: What did you change in `inst/resources/gitbook/css/style.css`?

**Review by yihui — APPROVED** [review]
Please also sign the Contributor License Agreement above, add a bullet in `NEWS.md`, and yourself as `ctb` in `DESCRIPTION`. Thanks a lot!

File: inst/resources/gitbook/css/style.css (hunk 1 above)
Comment (prdm0) [reply inst/resources/gitbook/css/style.css:9]: Yes, I had to change it to put the whatsapp symbol.

Comment (prdm0) [reply inst/resources/gitbook/css/style.css:9]: > Please also sign the Contributor License Agreement above, add a bullet in `NEWS.md`, and yourself as `ctb` in `DESCRIPTION`. Thanks a lot!

Ok, I will make the changes. Thanks.

**Comment by prdm0** [comment]
@yihui, everything is working properly and I am satisfied. I believe that the merge can be carried out. Thank you for the project and I am sure that this contribution will help many people.

Best regards.

File: inst/resources/gitbook/css/style.css (hunk 1 above)
Comment (yihui) [reply inst/resources/gitbook/css/style.css:9]: I'm not sure why the `whatsapp` symbol was not included before. I'll let @cderv review it. Thanks!

**Review by yihui — APPROVED** [review]
The JavaScript looks good to me. I'll let @cderv review the CSS to make sure the next time we upgrade FontAwesome, the symbol won't be lost by accident. Thanks!

File: inst/resources/gitbook/css/style.css (hunk 1 above)
Comment (prdm0) [reply inst/resources/gitbook/css/style.css:9]: @yihui, Exact, since the symbol was introduced in version v4.3 of Font Awesome.

**Comment by cderv** [comment]
Hi @prdm0 ! 

Just to be sure, in your last commit you are upgrading all fontawesome deps to 5.15.3 and not just add the fontawesome whatsapp icon ?

**Comment by prdm0** [comment]
> Hi @prdm0 !
> 
> Just to be sure, in your last commit you are upgrading all fontawesome deps to 5.15.3 and not just add the fontawesome whatsapp icon ?

Hi @cderv,

initially, I had just added the whatsapp icon to the inst / resources / gitbook / css / style.css file.

This works so that whatsapp is a sharing option.

I later realized that there was an incentive to use version 5.15.3 of FontAwesome <https://fontawesome.com/>. I noticed that several other icons, besides WhatsApp were available and were not in the old style.css.

In order not to load style.css with too much information, I used the summary file all.min.css (FontAwesome 5.15.3) as the basis for the update, that is, all olds icons plus some were added.

**Review by cderv — CHANGES_REQUESTED** [review]
Why did you remove `fontawesome-webfont.ttf` in your last commit ? 

You replaced it with css ? is that so ? 

Because the font is still used in the CSS. 

If I try on a demo, I don't have any icons anymore with your change. I think the ttf file is still needed.

This PR was just about adding whatapps and now it seems it is about updating fontawesome for `gitbook()` 

Can you clarify that for me before I do any change ?

Thanks.

**Comment by cderv** [comment]
The ttf file is still there: https://prdm0.github.io/regimento_interno/libs/gitbook-2.6.7/css/fontawesome/fontawesome-webfont.ttf and I think that is why you see the correct icon

> Maybe something went unnoticed by me.

Don't forget to run `clean_book(TRUE)` to completely remove all old component. It is highly possible that the ttf file was copied there from an old built. Rendering a new book with new version won't delete this file. You can check in your `_book/` folder (or equivalent output dir). 

I understand what you want to do. I'll take it from here to correctly update the fontawesome lib. 

It is not so easy to pass between version (4 to 5) and that is why we don't do that often. Future plans will be to use the new **fontawesome** package in all other places.

**Comment by prdm0** [comment]
> The ttf file is still there: https://prdm0.github.io/regimento_interno/libs/gitbook-2.6.7/css/fontawesome/fontawesome-webfont.ttf and I think that is why you see the correct icon
> 
> > Maybe something went unnoticed by me.
> 
> Don't forget to run `clean_book(TRUE)` to completely remove all old component. It is highly possible that the ttf file was copied there from an old built. Rendering a new book with new version won't delete this file. You can check in your `_book/` folder (or equivalent output dir).
> 
> I understand what you want to do. I'll take it from here to correctly update the fontawesome lib.
> 
> It is not so easy to pass between version (4 to 5) and that is why we don't do that often. Future plans will be to use the new **fontawesome** package in all other places.

Sorry @cderv, for some reason, using the same project in RStudio the file remains. I think that the RStudio project does not rebuild and takes advantage of what had already been built.

Sorry for that nonsense.

Anyway, considering the file again, everything works again, with more icons and with whatsapp working.

I will change. Grateful for the observation.

**Comment by prdm0** [comment]
Hi @cderv, thanks. Let me explain what is different about this last commit:

1 - style.css now has more symbols, including whatsapp. Everything that had before remains;
2 - Sharing on whatsapp is now allowed and works on mobile and desktop. Therefore, it is now possible to do something like:

```YAML
title: "The mystery"
author: "Sherlock Holmes"
site: bookdown::bookdown_site
output: 
  bookdown::gitbook:
    config:
      sharing:
        whatsapp: true
```

Thanks again.

**Comment by cderv** [comment]
Ok. After looking more into this Fontawesome update, I would prefer we do that cleanly if we upgrade from 4.7 to 5 

* https://fontawesome.com/how-to-use/on-the-web/setup/hosting-font-awesome-yourself
* https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4#manually

This would mean using the new `all.css` and new `webfonts` file (as you said ttf is not the same), and also insuring that shims for V4 to V5 is ok. 

So I would just do what you do at the beginning: add the whatapps icon only. As you said ([`12f7281` (#1125)](https://github.com/rstudio/bookdown/pull/1125/commits/12f72818534e61d08151b123352ff165c32a978b#r607823298)) it was introduced in 4.3 (https://github.com/FortAwesome/Font-Awesome/issues/11978) so it should be present in our repo. But it is not. I think previous update from 4.1 to 4.7 was not done correctly: https://github.com/rstudio/bookdown/pull/464

Are you ok to just add the CSS for whatapps icon ? It seems it is working fine with just this addition. 

We'll then work properly to update fontawesome deps later on.

**Comment by prdm0** [comment]
> Ok. After looking more into this Fontawesome update, I would prefer we do that cleanly if we upgrade from 4.7 to 5
> 
>     * https://fontawesome.com/how-to-use/on-the-web/setup/hosting-font-awesome-yourself
> 
>     * https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4#manually
> 
> 
> This would mean using the new `all.css` and new `webfonts` file (as you said ttf is not the same), and also insuring that shims for V4 to V5 is ok.
> 
> So I would just do what you do at the beginning: add the whatapps icon only. As you said ([`12f7281` (#1125)](https://github.com/rstudio/bookdown/pull/1125/commits/12f72818534e61d08151b123352ff165c32a978b#r607823298)) it was introduced in 4.3 ([FortAwesome/Font-Awesome#11978](https://github.com/FortAwesome/Font-Awesome/issues/11978)) so it should be present in our repo. But it is not. I think previous update from 4.1 to 4.7 was not done correctly: #464
> 
> Are you ok to just add the CSS for whatapps icon ? It seems it is working fine with just this addition.
> 
> We'll then work properly to update fontawesome deps later on.

I think it is wiser to do as you said. I will submit in a few minutes a new commit with the old sytile.css file, just adding the whatsapp icon.

**Comment by prdm0** [comment]
@cderv, it is now as it was before and no major changes to style.css have been made. Only the WhatsApp icon was added.

**Review by cderv — APPROVED** [review]
I pushed some slight addition and I isolated the whatapps icon so we know to check for it when we will upgrade. 

Thanks !

**Comment by prdm0** [comment]
> I pushed some slight addition and I isolated the whatapps icon so we know to check for it when we will upgrade.
> 
> Thanks !

Important so that it is not overlooked in future updates. Thanks.

**Pull request merged**